File: src/snippits/load_credentials_from_json_file_in_home_directory/load_credentials_from_json_file_in_home_directory.rb
#!/usr/bin/ruby

require 'json'
require 'getoptlong'
require 'pp'

opts = GetoptLong.new(
  [ '--ipam-credentials-file', GetoptLong::OPTIONAL_ARGUMENT ],
)

# pp(opts)

options = {
  'ipam-credentials-file' => File.expand_path('~/.config/IAS/some-awesome-project/ipam_credentials.json')
}

opts.each do |opt, arg|
  case opt
    when '--ipam-credentials-file'
      options['ipam-credentials-file'] = arg
  end
end

data=JSON.parse(File.read(options['ipam-credentials-file']))
print 'Username: ' + data['username'] + "\n"

File: src/snippits/load_credentials_from_json_file_in_home_directory/load_credentials_from_json_file_in_home_directory.pl
#!/usr/bin/perl

use strict;
use warnings;

use JSON;
use Data::Dumper;
use Getopt::Long;

my $OPTIONS_VALUES = {
	'ipam-credentials-file' => glob('~/.config/IAS/some-awesome-project/ipam_credentials.json'),
};

my @OPTIONS = (
	'ipam-credentials-file=s'
);

GetOptions(
	$OPTIONS_VALUES,
	@OPTIONS,
)
or die "Bad options.\n";

print Dumper(load_json_file($OPTIONS_VALUES->{'ipam-credentials-file'}));

exit;

sub load_json_file
{
	my ($filename) = @_;
	use IO::File;
	my $fh = IO::File->new($filename)
		or die "Can't open $filename for reading: $!";
	my $contents;
	{
		local $/;
		$contents = <$fh>;
	}
	$fh->close();
	return JSON::decode_json($contents);
}
